support-workers/src/test/scala/com/gu/zuora/ZuoraITSpec.scala

```diff
@@ -39,6 +40,24 @@ class ZuoraITSpec extends AsyncFlatSpec with Matchers {
     }
   }
 
+  it should "retrieve subscription redemption information from a redemption code" in {
+    val redemptionCode = "gd12-fnc4ox"
+    uatService.getSubscriptionFromRedemptionCode(RedemptionCode(redemptionCode).right.get).map {
+      response =>
+        response.records.size shouldBe 1
+        response.records.head.gifteeIdentityId shouldBe None
+        response.records.head.redemptionCode shouldBe Some(redemptionCode)
+    }
+  }
+
+  it should "handle invalid redemption codes" in {
+    val invalidRedemptionCode = "xxxx-0000"
+    uatService.getSubscriptionFromRedemptionCode(RedemptionCode(invalidRedemptionCode).right.get).map {
+      response =>
+        response.records.size shouldBe 0
+    }
+  }
+
   it should "be resistant to 'ZOQL injection'" in {
     // try https://github.com/guardian/support-service-lambdas/blob/master/lib/zuora/src/main/scala/com/gu/util/zuora/SafeQueryBuilder.scala
```

Comment:
unrelated, but that branch has been renamed now
```suggestion
    // try https://github.com/guardian/support-service-lambdas/blob/main/lib/zuora/src/main/scala/com/gu/util/zuora/SafeQueryBuilder.scala
```